design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,280 @@
+This function will parse the `xms_mirid` field of the Azure token provided by the Azure VM, extracting resource group, system/user assigned identity, and client ID. Once extracted, we will compare them to the annotations of the Conjur host for this Azure resource. If they match, the Azure VM resource will receive a Conjur access token in return. All further requests that this Azure VM sends will need to have this Conjur access token in the header.
+
+### Backwards compatibility
+
+None because this is a new feature.
+
+### Performance
+
+Azure authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+TBT
+
+## Logs
+
+|    | Scenario                                                              | Log message                                                                            | Notes                                                                                | Status | Exists?                                       |
+|--- |-----------------------------------------------------------------------|----------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|--------|-----------------------------------------------|
+| 1  | AuthenticatorNotWhitelisted (in DB/ENV)                               | {0-authenticator-name}' is not enabled                                                 |                                                                                      |        | Yes                                           |
+| 2  | WebserviceNotFound (in Conjur)                                        | Webservice '{0-webservice-name}' wasn't found                                          |                                                                                      |        | Yes                                           |
+| 3  | WebserviceNotFound (in Conjur)                                        | '{0-role-name}' does not have 'authenticate' privilege on {1-service-name}             |                                                                                      |        | Yes                                           |
+| 4  | RoleNotFound (Host/User not defined in Conjur)                        | '{0-role-name}' wasn't found                                                           |                                                                                      |        | Yes                                           |
+| 5  | ProviderDiscoveryTimeout (couldn't make connection with provider-uri) | Azure Identity Provider failed with timeout error (Provider URI: '{0}'). Reason: '{1}' |                                                                                      |        | Yes (needed to be generalized)                |
```

Comment:
Maybe Azure authenticator?

Reply:
This timeout would be raised as a result of inability to talk to the `provider-uri` (https://sts.windows.net/TENANT_ID) so I don't think it would be a timeout of our Azure authn, rather the actual tenant Identity Provider. What do you think?

Reply:
@sigalsax Thank you for this clarification. I agree.